### DISCUSSION TITLE
Error: required external tools are missing:  - npm CLI is not installed, please see https://nodejs.org/ to install

### DISCUSSION THREAD

**Discussion by siegfried01** [Azure/azure-dev#536] [Q&A] [answered]
I'm trying to follow along at with the demo at [https://www.youtube.com/watch?v=LjI-WZtunqA](url) .

After having updated azure cli & installing adzI do 

    azd init

And then I select the 

    todo-python-mongo

option and then right click azure.yml and select

     up

option and I get the error

     Error: required external tools are missing: - npm CLI is not installed, please see https://nodejs.org/ to install

I had npm installed and when I received the above error I installed the latest version (16.17.0 LTS) and npm/node.

This did not help! I received the same error.

I tried the azd cosmos C# todo option and received that same error again.

I had installed azure-cli 2.39 before starting this exercise and  azd version  version 0.1.0-beta.5 (commit 7bd25dd6f1b1899abf63b5f6f36d268a465183e4)

I'm running windows 10 & pwsh 7.2.6

I installed 16.17.0 LTS node/npm from here: [https://nodejs.org/en/](url) after my initial attempt failed.

This adz looks like a very nice tool... I would love to see it work on my personal dev machine.

Thanks

Siegfried

**Comment by savannahostrowski** [comment, marked as answer]
Hey there - can you try restarting your terminal to make sure that node made it into your PATH and then try to run `azd up` again?

What does `npm --version` return?

**Reply by siegfried01** [reply]
It is working now.... maybe the new version fixed it.
